### ISSUE TITLE
Spike: Research why deployments take too long

### ISSUE THREAD

**Issue by hmstepanek** [hypothesis/product-backlog#896]
From the time of a merge to the change taking effect on production, it takes ~40mins. This issue is a spike to research and figure out why this is taking so long.

**Comment by robertknight** [comment]
@dmfine - It looks to me as though this problem is sufficiently "solved" for h that we can close this now. Could you post some notes in a comment here to clarify:

- What changes were made to improve the situation?
- Are any caveats that developers should be aware of when they deploy changes in future?
- Do we still need to apply the changes to other projects (if yes and you're happy it is safe to do so, please just go ahead and do it)?
- Is there any more work planned for future?

**Comment by dmfine** [comment]
By watching the jenkins console I determined that most of the delay in the "deploy" step was due to Elastic Beanstalk using the "immutable" deployment strategy. It would test the new code for 2 minutes on a single instance, spin up the remainder, and test again. Instead I switched to "rolling updates with batch" with batch size of 100%. I also disable "Enhanced healthchecks" for the QA stage to shave off that 2 minutes. 
I will proceed to make this change for our other builds. 
There is more work to do, for instance switching to only python3 would speed up the testing step a lot. Hannah's work on optimizing the dockerfile will help. And ultimately we'll move to a deployment process with pre-baked AMIs made with packer which will reduce the time of installing dependencies quite a lot.
